File: q3.rb
#CSI3130 - Assignment4 - Part1
#Question 3: Estimate the value of pi in Ruby using random points

iters = 1000000.0 #Number of iterations to run
radius = 1.0
count = 0; #Keep track of number of points in the circle

for i in 0..(iters - 1)
    x = rand(0..1);
    y = rand(0..1);
    coords = (x * x) + (y * y);
    #If coordinates are within circle, increment count
    if (coords <= radius * radius)
        count = count + 1
    end
end
#Estimate pi as the count within the circle divided by the count in the square(which is all points, so # of iterations)
pi = (count * 4.0) / iters;

puts("The number of points in the square is: " + iters.to_s)
puts("The number of points inside the circle is: " + count.to_s)
puts("The estimation of pi is: " + pi.to_s)

File: q3.php
<?php
//CSI3130 - Assignment4 - Part1
//Question 3: Estimate the value of pi in PHP using random points

$iters         = 1000000.0; //Number of iterations to run
$radius        = 1.0;
$count         = 0; //Keep track of the number of points in the circle

for ($i = 0; $i < $iters; $i++) {
    $x = rand(0, 1); //Generate random number between 0 and 1
    $y = rand(0, 1); //Generate random number between 0 and 1

    $coords = ($x * $x) + ($y * $y);
    #If coordinates are within circle, increment count
    if ($coords <= ($radius * $radius)) {
        $count++;
    }
}
#Estimate pi as the count within the circle divided by the count in the square(which is all points, so # of iterations)
$pi = ($count * 4.0) / $iters;

echo "The number of points in the square is: ";
echo $iters;
echo "\n";
echo "The number of points inside the circle is: ";
echo $count;
echo "\n";
echo "The estimatation of pi is: ";
echo $pi;
echo "\n";
?>